connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/agent/DefaultAiAgentRequestHandler.java

```diff
@@ -0,0 +1,158 @@
+
+  public DefaultAiAgentRequestHandler(
+      ObjectMapper objectMapper,
+      ChatModelFactory chatModelFactory,
+      ToolCallingHandler toolCallingHandler) {
+    this.objectMapper = objectMapper;
+    this.chatModelFactory = chatModelFactory;
+    this.toolCallingHandler = toolCallingHandler;
+  }
+
+  @Override
+  public AgentResponse handleRequest(OutboundConnectorContext context, AgentRequest request) {
+    final AgentRequest.AgentRequestData requestData = request.data();
+    final AgentContext agentContext =
+        Optional.ofNullable(requestData.context()).orElseGet(AgentContext::empty);
+
+    // initialize configured model
+    final ChatLanguageModel chatModel = chatModelFactory.createChatModel(request);
+
+    // set up memory and load from context if available
+    final AgentContextChatMemoryStore chatMemoryStore =
+        new AgentContextChatMemoryStore(objectMapper);
+    chatMemoryStore.loadFromAgentContext(agentContext);
+
+    final ChatMemory chatMemory =
+        MessageWindowChatMemory.builder()
+            .maxMessages(
+                Optional.ofNullable(requestData.memory().maxMessages())
+                    .orElse(DEFAULT_MAX_MEMORY_MESSAGES))
+            .chatMemoryStore(chatMemoryStore)
+            .build();
+
+    // check configured guardrails
+    checkGuardrails(requestData, agentContext);
+
+    // update memory with system + new user messages/tool call responses
+    addSystemPromptIfNecessary(chatMemory, requestData);
+    addUserMessagesFromRequest(agentContext, chatMemory, requestData);
+
+    // call LLM API with updates messages + resolved tool specifications
+    final var toolSpecifications =
+        toolCallingHandler.loadToolSpecifications(
+            context.getJobContext().getProcessDefinitionKey(),
+            requestData.tools().containerElementId());
+
+    final var chatRequest =
+        ChatRequest.builder()
+            .messages(chatMemory.messages())
+            .toolSpecifications(toolSpecifications)
+            .build();
+
+    final ChatResponse chatResponse = chatModel.chat(chatRequest);
+    final AiMessage aiMessage = chatResponse.aiMessage();
+    chatMemory.add(aiMessage);
+
+    // extract tool call requests from LLM response
+    final var toolsToCall = toolCallingHandler.extractToolsToCall(toolSpecifications, aiMessage);
+    final var nextAgentState =
+        !toolsToCall.isEmpty() ? AgentState.WAITING_FOR_TOOL_INPUT : AgentState.READY;
+
+    // update context
+    final var updatedContext =
+        chatMemoryStore
+            .storeToAgentContext(agentContext)
+            .withState(nextAgentState)
+            .withMetrics(
+                agentContext
+                    .metrics()
+                    .incrementModelCalls(1)
+                    .incrementTokenUsage(AgentMetrics.TokenUsage.from(chatResponse.tokenUsage())));
+
+    return new AgentResponse(updatedContext, updatedContext.memory().getLast(), toolsToCall);
```

Comment:
This will be emitted as process variable - here it is crucial that we re-inject it on for the next call. This is a manual configuration for now, but we'll work on making this easier in the future. 

![image](https://github.com/user-attachments/assets/60d69097-be59-40c8-aea0-990d4f7c5035)